### PR TITLE
ApplicationAutoscaling: support for all the current various forms of resource_id

### PR DESCRIPTION
**Pull request by jweite**
Re Issue #3304.

### CODE REVIEW

File: moto/applicationautoscaling/models.py [hunk 1]
```diff
@@ -135,7 +135,17 @@ def _target_params_are_valid(namespace, r_id, dimension):
         try:
             valid_dimensions = [d.value for d in ScalableDimensionValueSet]
             d_namespace, d_resource_type, scaling_property = dimension.split(":")
-            resource_type, cluster, service = r_id.split("/")
+            if r_id.startswith("arn:aws:comprehend"):
```
Comment (bblommers) [inline moto/applicationautoscaling/models.py:138]: Can we extract this into a separate method, and add some comments to make it clear what r_id should look like?

**Review by bblommers — CHANGES_REQUESTED** [review]
Thanks for the PR @jweite - just have a single comment.

File: moto/applicationautoscaling/models.py (hunk 1 above)
Comment (jweite) [reply moto/applicationautoscaling/models.py]: Done.  I think this build is OK - it's the recurring batch issue.  It built fine on my Travis.

**Review by bblommers — APPROVED** [review]
LGTM

**Pull request merged**